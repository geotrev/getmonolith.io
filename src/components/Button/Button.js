import React from "react"

function getTag(href) {
  return href ? "a" : "button"
}

function getType(href, type) {
  return href || !type ? null : type
}

function getDisabledStatus(disabled, href) {
  const warnMsg = "*** You can't use a `disabled` state on anchor tags ***"
  if (disabled) {
    return href ? console.warn(warnMsg) : disabled
  }
}

export default function Button({
  href,
  disabled,
  type,
  id,
  tabIndex,
  onClick,
  onBlur,
  onFocus,
  className,
  dataParent,
  dataTarget,
  children,
}) {
  const Tag = getTag(href)
  return (
    <Tag
      data-parent={dataParent}
      data-target={dataTarget}
      disabled={getDisabledStatus(disabled, href)}
      className={className}
      onFocus={onFocus}
      onBlur={onBlur}
      onClick={onClick}
      href={href}
      type={getType(href, type)}
      tabIndex={tabIndex}
      id={id}
    >
      {children}
    </Tag>
  )
}
