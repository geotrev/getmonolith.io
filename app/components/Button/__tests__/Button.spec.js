import React from "react"
import Button from "../Button"

describe("<Button />", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<Button>Test</Button>)
    expect(wrapper).toMatchSnapshot()
  })

  it("renders with button tag by default", () => {
    const wrapper = shallow(<Button>Test</Button>)
    expect(wrapper.is("button")).toEqual(true)
  })

  it("renders an anchor if this.props.href is received", () => {
    const wrapper = shallow(<Button href="#">Test</Button>)
    expect(wrapper.is("a")).toEqual(true)
  })

  it("button tag can receive disabled state with this.props.disabled", () => {
    const wrapper = shallow(<Button disabled>Test</Button>)
    expect(wrapper.prop("disabled")).toBeDefined()
  })

  it("calls console.warn if link button has disabled attribute", () => {
    console.warn = jest.fn()
    mount(
      <Button href="#" disabled>
        Test
      </Button>
    )
    expect(console.warn).toHaveBeenCalledWith(
      "*** You can't use a `disabled` state on anchor tags ***"
    )
  })

  it("renders a 'submit' button if [type='submit']", () => {
    const wrapper = shallow(<Button type="submit">Test</Button>)
    expect(wrapper.prop("type")).toEqual("submit")
  })
})
