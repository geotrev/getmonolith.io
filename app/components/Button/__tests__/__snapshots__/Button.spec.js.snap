// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders 1`] = `
<button
  type={null}
>
  Test
</button>
`;
